plugins/modules/cloud/misc/proxmox_disk.py

```diff
@@ -0,0 +1,726 @@
+    type: bool
+  trans:
+    description:
+      - Force disk geometry bios translation mode.
+    type: str
+    choices: ['auto', 'lba', 'none']
+  werror:
+    description:
+      - Write error action.
+    type: str
+    choices: ['enospc', 'ignore', 'report', 'stop']
+  wwn:
+    description:
+      - The drive's worldwide name, encoded as 16 bytes hex string, prefixed by C(0x).
+    type: str
+extends_documentation_fragment:
+  - community.general.proxmox.documentation
+'''
+
+EXAMPLES = '''
+- name: Create new disk in VM (do not rewrite in case it exists already)
+  community.general.proxmox_disk:
+    api_host: node1
+    api_user: root@pam
+    api_token_id: token1
+    api_token_secret: some-token-data
+    name: vm-name
+    disk: scsi3
+    backup: True
+    cache: none
+    storage: local-zfs
+    size: 5
+    state: present
+
+- name: Create new disk in VM (force rewrite in case it exists already)
+  community.general.proxmox_disk:
+    api_host: node1
+    api_user: root@pam
+    api_token_id: token1
+    api_token_secret: some-token-data
+    vmid: 101
+    disk: scsi3
+    format: qcow2
+    storage: local
+    size: 16
+    force_replace: True
+    state: present
+
+- name: Update existing disk
+  community.general.proxmox_disk:
+    api_host: node1
+    api_user: root@pam
+    api_token_id: token1
+    api_token_secret: some-token-data
+    vmid: 101
+    disk: ide0
+    backup: False
+    ro: True
+    aio: native
+    state: updated
+
+- name: Grow existing disk
+  community.general.proxmox_disk:
+    api_host: node1
+    api_user: root@pam
+    api_token_id: token1
+    api_token_secret: some-token-data
+    vmid: 101
+    disk: sata4
+    size: +5G
+    state: grown
+
+- name: Detach disk (leave it unused)
+  community.general.proxmox_disk:
+    api_host: node1
+    api_user: root@pam
+    api_token_id: token1
+    api_token_secret: some-token-data
+    name: vm-name
+    disk: virtio0
+    state: detached
+
+- name: Move disk to another storage
+  community.general.proxmox_disk:
+    api_host: node1
+    api_user: root@pam
+    api_password: secret
+    vmid: 101
+    disk: scsi7
+    target_storage: local
+    format: qcow2
+    state: moved
+
+- name: Move disk from one VM to another
+  community.general.proxmox_disk:
+    api_host: node1
+    api_user: root@pam
+    api_token_id: token1
+    api_token_secret: some-token-data
+    vmid: 101
+    disk: scsi7
+    target_vmid: 201
+    state: moved
+
+- name: Remove disk permanently
+  community.general.proxmox_disk:
+    api_host: node1
+    api_user: root@pam
+    api_password: secret
+    vmid: 101
+    disk: scsi4
+    state: absent
+'''
+
+RETURN = '''
+vmid:
+  description: The VM vmid.
+  returned: success
+  type: int
+  sample: 101
+msg:
+  description: A short message
+  returned: always
+  type: str
+  sample: "Disk scsi3 created in VM 101"
+'''
+
+from ansible.module_utils.basic import AnsibleModule, env_fallback
```

Comment:
Looks like `env_fallback` is not used anywhere. If so, please remove it.

Reply:
You're right, fixed.